Tools/RemoteSynchronization/Program.cs

```diff
@@ -0,0 +1,815 @@
+            if (renamed > 0)
+                Duplicati.Library.Logging.Log.WriteInformationMessage(LOGTAG, "rsync", "Renamed {0} files in {1}", renamed, b2s.DisplayName);
+
+            Duplicati.Library.Logging.Log.WriteInformationMessage(LOGTAG, "rsync", "Remote synchronization completed successfully");
+
+            return 0;
+        }
+
+        // TODO have concurrency parameters: uploaders, downloaders
+        // TODO low memory mode, where things aren't kept in memory. Maybe utilize SQLite?
+        // TODO For convenience, have the option to launch a "duplicati test" on the destination backend after the synchronization
+        // TODO Save hash to minimize redownload
+        // TODO Duplicati Results
+
+        /// <summary>
+        /// Copies the files from one backend to another.
+        /// The files are copied one by one, and each file is verified after uploading if the verify flag is set.
+        /// </summary>
+        /// <param name="b_src">The source backend.</param>
+        /// <param name="b_dst">The destination backend.</param>
+        /// <param name="files">The files that will be copied.</param>
+        /// <param name="dry_run">Flag for whether destructive actions (writes) should be printed rather than performed. Downloads (reads) are still being performed.</param>
+        /// <param name="verify">Flag for whether to verify each upload to the destination backend.</param>
+        /// <returns>A tuple holding the number of succesful copies and a List of the files that failed.</returns>
+        private static async Task<(long, IEnumerable<IFileEntry>)> CopyAsync(IStreamingBackend b_src, IStreamingBackend b_dst, IEnumerable<IFileEntry> files)
+        {
+            long successful_copies = 0;
+            List<IFileEntry> errors = [];
+            using var s_src = new MemoryStream();
+            using var s_dst = new MemoryStream();
+            long i = 0, n = files.Count();
+            var sw_get_src = new System.Diagnostics.Stopwatch();
+            var sw_put_dst = new System.Diagnostics.Stopwatch();
+            var sw_get_dst = new System.Diagnostics.Stopwatch();
+            var sw_get_cmp = new System.Diagnostics.Stopwatch();
+
+            foreach (var f in files)
+            {
+                if (GlobalConfig.Progress)
+                    Console.Write($"\rCopying: {i}/{n}");
+
+                Duplicati.Library.Logging.Log.WriteVerboseMessage(LOGTAG, "rsync", "Copying {0} from {1} to {2}", f.Name, b_src.DisplayName, b_dst.DisplayName);
+
+                try
+                {
+                    sw_get_src.Start();
+                    await b_src.GetAsync(f.Name, s_src, CancellationToken.None);
+                    sw_get_src.Stop();
+                    if (GlobalConfig.DryRun)
+                    {
+                        Duplicati.Library.Logging.Log.WriteDryrunMessage(LOGTAG, "rsync", "Would write {0} bytes of {1} to {2}", s_src.Length, f.Name, b_dst.DisplayName);
```

Comment:
There is a human-size formatter in `Duplicati.Library.Utility.Utility.FormatSizeString`